lib/src/findex/findex.dart

```diff
@@ -115,8 +125,21 @@ class Findex {
       );
       final end = DateTime.now();
 
+      if (errorCode != 0 &&
+          outputLengthPointer.value > defaultOutputSizeInBytes) {
+        return upsert(masterKey, label, additions, deletions, fetchEntries,
+            upsertEntries, insertChains,
+            outputSizeInBytes: outputLengthPointer.value);
+      }
```

Comment:
You should not upsert a second time.

Reply:
as discussed, the first call to `h_upsert` is only used to get the allocation size to use in the 2nd call